service/storage_proxy.cc

```diff
@@ -191,6 +191,27 @@ template <> struct fmt::formatter<service::storage_proxy_remote_read_verb> : fmt
 
 namespace service {
 
+static lw_shared_ptr<query::read_command> reversed(lw_shared_ptr<query::read_command>&& cmd)
+{
+    auto schema = local_schema_registry().get(cmd->schema_version)->get_reversed();
+    cmd->schema_version = schema->version();
+    cmd->slice = query::legacy_reverse_slice_to_native_reverse_slice(*schema, cmd->slice);
+
+    return cmd;
```

Comment:
You need `return std::move(cmd)` here to move `cmd`, because unintuitively, a `&&` which has a name, is not a candidate for move.

Reply:
so why pass `cmd` by rvalue ref?  can be passed by value just as well.

Reply:
It was initially passed by value and I was asked to make it rvalue. That's a minor thing.

Reply:
In general, using an rvalue is only needed when you want to force ownership change. It is best to prefer value params in the general case.